src/data/list/basic.lean

```diff
@@ -39,6 +39,10 @@ theorem tail_eq_of_cons_eq {h₁ h₂ : α} {t₁ t₂ : list α} :
       (h₁::t₁) = (h₂::t₂) → t₁ = t₂ :=
 assume Peq, list.no_confusion Peq (assume Pheq Pteq, Pteq)
 
+theorem cons_ne_self {α : Type u} : ∀ (a : α) (l : list α), a::l ≠ l
```

Comment:
Thoughts on tagging this `@[simp]`?

Reply:
I don't think it should be `@[simp]`. It will not be indexed very well, and I also don't know many proofs that would be assisted by its presence. Do you have a use case for this theorem?

Reply:
Also, any reason you don't like the proof I gave on zulip? I don't see the need to do a proof by induction here when it follows from `succ_ne_self`

Reply:
Because `length` isn't defined for another 130 lines, but I want to `cons_ne_self` to appear very close to `cons_ne_nil`

Reply:
> Do you have a use case for this theorem?

Not really, it came up in a lemma about `prod.lex`, but frankly naming it explicitly would have been fine.

Reply:
I would suggest just moving it down.